griddb/src/main/java/com/yahoo/ycsb/db/GridDBClient.java

```diff
@@ -0,0 +1,331 @@
+ * Copyright (c) 2018 YCSB contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+ 
+package com.yahoo.ycsb.db;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+
+import com.toshiba.mwcloud.gs.ColumnInfo;
+import com.toshiba.mwcloud.gs.Container;
+import com.toshiba.mwcloud.gs.ContainerInfo;
+import com.toshiba.mwcloud.gs.ContainerType;
+import com.toshiba.mwcloud.gs.GSException;
+import com.toshiba.mwcloud.gs.GSType;
+import com.toshiba.mwcloud.gs.GridStore;
+import com.toshiba.mwcloud.gs.GridStoreFactory;
+import com.toshiba.mwcloud.gs.PartitionController;
+import com.toshiba.mwcloud.gs.Row;
+import com.yahoo.ycsb.ByteArrayByteIterator;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+
+/**
+ * A class representing GridDBClient.
+ *
+ */
+public class GridDBClient extends com.yahoo.ycsb.DB {
+  //(A)multicast method
+  private String notificationAddress = ""; // "239.0.0.1"
+  private String notificationPort = ""; // "31999"
+  //(B)fixed list method
+  private String notificationMember = ""; // "10.0.0.12:10001,10.0.0.13:10001,10.0.0.14:10001"
+
+  private String userName = ""; //
+  private String password = ""; //
+  private String clusterName = ""; // "ycsbCluster"
+
+
+  public static final String VALUE_COLUMN_NAME_PREFIX= "field";
+  public static final int ROW_KEY_COLUMN_POS = 0;
+  public static final String CONTAINER_PREFIX = "usertable@";
+
+  public static final int DEFAULT_CACHE_CONTAINER_NUM = 1000;
+  public static final int FIELD_NUM = 10;
+
+  private int numContainer = 0; // Sets PartitionNum
+  public static final GSType SCHEMA_TYPE = GSType.STRING;
+
+  private GridStore store;
+  private ContainerInfo containerInfo = null;
+
+  class RowComparator implements Comparator<Row> {
+    public int compare(Row row1, Row row2) throws NullPointerException {
+      int result = 0;
+      try {
+        Object val1 = row1.getValue(0);
+        Object val2 = row2.getValue(0);
+        result = ((String)val1).compareTo((String)val2);
+      } catch (GSException e) {
+        e.printStackTrace();
+        throw new NullPointerException();
+      }
+      return result;
+    }
+  }
+
+  public void init() throws DBException {
+    System.out.println("GridDBClient");
```

Comment:
please use a logger instead of System.out

Reply:
I reflected source code using a logger by 8bfda27d1f97f9e56bc24f949373ecb8eb98ee05 .